### ISSUE TITLE
[Gun Bonsai] Swiftness upgrade breaks sound in GZDoom 4.11.3

### ISSUE THREAD

**Issue by cubebert-github** [ToxicFrog/doom-mods#27]
In Gun Bonsai 0.10.5, the swiftness upgrade can cause the majority of sounds to not play anymore until the player kills another monster with the weapon that has the upgrade. The bug can be triggered by either activating the menu or the console while the world freezes after getting a kill with the upgrade enabled. Afterwards, activating the menu or the console again will cause almost every sound to not play, excluding the item pickup sound and any sounds from the menu. This bug persists between every level, as while it will restore the sound, activating menus will cause it to break again. I've encountered this bug with the latest version of GZDoom and Gun Bonsai, and while the video that shows the bug uses Eviternity II, I believe the bug can occur without any other mods.

https://github.com/ToxicFrog/doom-mods/assets/24487663/b108b6bb-fe9f-4d47-8fc4-e38680cb5b9b

**Comment by ToxicFrog** [comment]
I encountered this (or at least, a closely related bug causing sound to go away due to swiftness/menu interactions) while testing it, and made some changes specifically to fix it, and apparently that was not enough to fix it in the general case. I'll investigate.

**Comment by ToxicFrog** [comment]
Initial notes:
- this affects 4.11.1 as well
- this affects Indestructable in both time stop and slomo mode as well
- it is not necessary to open the console or menu while timestopped; merely triggering a timestop and then opening and closing the console after it wears off will turn off sounds
- it does not affect the vanilla `PowerTimeStop` powerup

**Comment by ToxicFrog** [comment]
This appears to be related to the code I added to mitigate the bad interactions between Swiftness, Indestructable, and other mods that stop or slow time like Gearbox and Bullet Time.

Looking at PowerTimeFreezer some more, I think it does not properly handle the case where the player has multiple subclasses of it in their inventory. So what is happening (with the bad interaction, not the sound bug) is something like:
- two timefreezers go off at once
- the player gets the `timefreezer` field set, making them immune to the effect while the rest of the level freezes
- one of the freezers expires
- the `timefreezer` field gets cleared, so the player is no longer immune to the level being frozen, but they still have a time freezer in their inventory
- the player is now frozen along with everything else
- RIP

The workaround I added fixed this, but broke some stuff related to other cleanup when a timefreezer expires, which presumably confuses the code that's meant to pause and unpause the SFX when wiggling the console.
